packages/esm-form-engine-app/src/form-renderer/form-renderer.component.tsx

```diff
@@ -55,6 +72,97 @@ const FormRenderer: React.FC<FormRendererProps> = ({
     [promptBeforeClosing],
   );
 
+  useEffect(() => {
+    if (typeof onBeforeGetContentResolve.current === 'function') {
+      onBeforeGetContentResolve.current();
+    }
+  }, [onBeforeGetContentResolve]);
+
+  const patientDetails = useMemo(() => {
+    if (!patient) {
+      return {};
+    }
+
+    const getGender = (gender: string): string => {
+      switch (gender) {
+        case 'male':
+          return t('male', 'Male');
+        case 'female':
+          return t('female', 'Female');
+        case 'other':
+          return t('other', 'Other');
+        case 'unknown':
+          return t('unknown', 'Unknown');
+        default:
+          return gender;
+      }
+    };
+
+    const identifiers =
+      patient.identifier?.filter(
+        (identifier) => !excludePatientIdentifierCodeTypes?.uuids.includes(identifier.type.coding[0].code),
+      ) ?? [];
+
+    return {
+      age: age(patient.birthDate),
+      gender: getGender(patient.gender),
+      identifiers: [...identifiers],
+      location: patient?.address?.[0].city,
+      name: patient ? getPatientName(patient) : '',
+    };
+  }, [excludePatientIdentifierCodeTypes?.uuids, patient, t]);
+
+  const handleOnBeforeGetContent = useCallback(
+    () =>
+      new Promise<void>((resolve) => {
+        if (patient && schema) {
+          setIsPrinting(true);
+          onBeforeGetContentResolve.current = resolve;
+
+          setTimeout(() => {
+            setIsPrinting(false);
+            resolve();
+          }, 2000);
```

Comment:
Why the 2000ms delay? Should it be fixed? What happens when you do something like:

```ts
      requestAnimationFrame(() => {
          setIsPrinting(false);
          resolve();
      });

```